state/protocol/badger/state.go

```diff
@@ -509,23 +487,29 @@ func bootstrapStatePointers(root protocol.Snapshot) func(*transaction.Tx) error
 	}
 }
 
-// bootstrapEpoch bootstraps the protocol state database with information about
-// the previous, current, and next epochs as of the root snapshot.
-func bootstrapEpoch(
+// bootstrapEpochForProtocolStateEntry bootstraps the protocol state database with epoch
+// information (in particular, EpochSetup and EpochCommit service events) associated with
+// a particular Dynamic Protocol State entry.
+// There may be several such entries within a single root snapshot, in which case this
+// function is called once for each entry. Entries may overlap in which underlying
+// epoch information (service events) they reference, which case duplicate writes of
+// the same data are ignored.
+func bootstrapEpochForProtocolStateEntry(
```

Comment:
Not sure about correctness of this function. Specifically the fact we store only setup/commit events for current, previous and next epochs, but we don't store them for all protocol entries. Previously you have made an example where such situation can occur in the sealing segment, naturally when node bootstraps it needs to have that data otherwise the won't be able to reconstruct the `RichProtocolStateEntry`

Reply:
Right, confused with old logic.